functions/src/stages/profile.utils.ts

```diff
@@ -57,35 +59,60 @@ export async function completeProfile(
   ) {
     return;
   }
+  const agentConfig = participant.agentConfig;
 
   // Fetch experiment creator's API key.
   const experimenterData =
     await getExperimenterDataFromExperiment(experimentId);
   if (!experimenterData) return null;
 
-  const response = await getAgentResponse(
+  const promptConfig: BasePromptConfig = {
+    id: stageConfig.id,
+    type: stageConfig.kind,
+    prompt: createDefaultPromptFromText(PROFILE_DEFAULT_PROMPT, stageConfig.id),
+    generationConfig: createModelGenerationConfig(),
+    structuredOutputConfig: PROFILE_STRUCTURED_OUTPUT_CONFIG,
+    numRetries: 0,
+  };
+  const structuredPrompt = await getStructuredPrompt(
+    experimentId,
+    participant.currentCohortId,
+    /*participantIds=*/ [participant.privateId],
+    stageConfig.id,
+    participant,
+    agentConfig,
+    promptConfig,
+  );
+
+  const response = await processModelResponse(
+    experimentId,
+    participant.currentCohortId,
+    /*participantId=*/ participant.privateId,
```

Comment:
Nit: cleanup?

Reply:
The `/*participantId=*/` is a [parameter-name comment](https://google.github.io/styleguide/tsguide.html#comments-when-calling-a-function) to make it clear what parameter this is going to.